homeassistant/components/homekit/accessories.py

```diff
@@ -85,8 +100,32 @@ def update_state_callback(self, entity_id=None, old_state=None,
         _LOGGER.debug('New_state: %s', new_state)
         if new_state is None:
             return
+        self.hass.async_add_job(self.update_battery_level, new_state)
         self.hass.async_add_job(self.update_state, new_state)
 
+    def update_battery_level(self, new_state):
+        """Update battery service if available."""
+        battery_level = convert_to_float(
+            new_state.attributes.get(ATTR_BATTERY_LEVEL))
+        if battery_level is None:
+            return
+        self.char_battery.set_value(battery_level)
+        self.char_low_battery.set_value(battery_level < 20)
+        _LOGGER.debug('%s: Updated battery level to %d', self.entity_id,
+                      battery_level)
+        self.hass.async_add_job(self.update_battery_charging, new_state)
+
+    def update_battery_charging(self, new_state):
+        """Update battery service if available."""
+        charging = new_state.attributes.get(ATTR_BATTERY_CHARGING)
+        if charging is None:
+            hk_charging = 2
```

Comment:
If an entity is not chargeable the first time, it won't be any time later. Maybe another flag?